Makefile

```diff
@@ -76,55 +78,70 @@ mocks:
 e2e: e2e-compose e2e-compose-standalone ## Run end to end local tests in both modes. Set E2E_TEST=TestName to run a single test
 
 .PHONY: build-and-e2e
-build-and-e2e: compose-plugin e2e-compose e2e-compose-standalone ## Compile the compose cli-plugin and run end to end local tests in both modes. Set E2E_TEST=TestName to run a single test
+build-and-e2e: build e2e-compose e2e-compose-standalone ## Compile the compose cli-plugin and run end to end local tests in both modes. Set E2E_TEST=TestName to run a single test
 
 .PHONY: cross
-cross: ## Compile the CLI for linux, darwin and windows
-	$(BUILDX_CMD) bake binary
+cross: buildx-cmd ## Compile the CLI for linux, darwin and windows
+	$(BUILDX_CMD) bake binary-cross
 
 .PHONY: test
-test: ## Run unit tests
+test: buildx-cmd ## Run unit tests
 	$(BUILDX_CMD) bake test
 
 .PHONY: cache-clear
-cache-clear: ## Clear the builder cache
+cache-clear: buildx-cmd ## Clear the builder cache
 	$(BUILDX_CMD) prune --force --filter type=exec.cachemount --filter=unused-for=24h
 
 .PHONY: lint
-lint: ## run linter(s)
+lint: buildx-cmd ## run linter(s)
 	$(BUILDX_CMD) bake lint
 
 .PHONY: docs
-docs: ## generate documentation
+docs: buildx-cmd ## generate documentation
 	$(eval $@_TMP_OUT := $(shell mktemp -d -t compose-output.XXXXXXXXXX))
 	$(BUILDX_CMD) bake --set "*.output=type=local,dest=$($@_TMP_OUT)" docs-update
 	rm -rf ./docs/internal
 	cp -R "$($@_TMP_OUT)"/out/* ./docs/
 	rm -rf "$($@_TMP_OUT)"/*
 
 .PHONY: validate-docs
-validate-docs: ## validate the doc does not change
+validate-docs: buildx-cmd ## validate the doc does not change
 	$(BUILDX_CMD) bake docs-validate
 
 .PHONY: check-dependencies
 check-dependencies: ## check dependency updates
 	go list -u -m -f '{{if not .Indirect}}{{if .Update}}{{.}}{{end}}{{end}}' all
 
 .PHONY: validate-headers
-validate-headers: ## Check license header for all files
+validate-headers: buildx-cmd ## Check license header for all files
 	$(BUILDX_CMD) bake license-validate
 
 .PHONY: go-mod-tidy
-go-mod-tidy: ## Run go mod tidy in a container and output resulting go.mod and go.sum
+go-mod-tidy: buildx-cmd ## Run go mod tidy in a container and output resulting go.mod and go.sum
 	$(BUILDX_CMD) bake vendor-update
 
 .PHONY: validate-go-mod
-validate-go-mod: ## Validate go.mod and go.sum are up-to-date
+validate-go-mod: buildx-cmd ## Validate go.mod and go.sum are up-to-date
 	$(BUILDX_CMD) bake vendor-validate
 
 validate: validate-go-mod validate-headers validate-docs ## Validate sources
 
-pre-commit: validate check-dependencies lint compose-plugin test e2e-compose
+pre-commit: validate check-dependencies lint build test e2e-compose
+
+.PHONY: buildx-cmd
+ifneq (, $(BUILDX_BIN))
+buildx-cmd:
+	$(eval BUILDX_CMD = $(BUILDX_BIN))
+else ifneq (, $(shell docker buildx version 2>/dev/null))
+buildx-cmd:
+	$(eval BUILDX_CMD = docker buildx)
+else ifneq (, $(shell which buildx 2>/dev/null))
+buildx-cmd:
+	$(eval BUILDX_CMD = $(shell which buildx))
```

Comment:
note that generally `command -v` is preferred over `which` (for portability). For the detection here, I think we would only need to know _IF_ it's available (and no need to run it using the full path, as we already use the `PATH` to discover)